### PR TITLE
App Check service config

### PR DESCRIPTION
**Pull request by rainshen49**
Add App Check Service Config resource. This is part of https://github.com/hashicorp/terraform-provider-google/issues/17095

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_firebase_app_check_service_config`
```

### CODE REVIEW

File: mmv1/templates/terraform/custom_create/firebase_app_check_service_config.go
Comment (rileykarson) [inline mmv1/templates/terraform/custom_create/firebase_app_check_service_config.go:27]: We should be able to append the update mask to the create and delete urls in order to avoid needing to use `custom_x` methods here. They're pretty heavyweight and can be hard to spot the difference between generated and custom/handwritten code, so we avoid them as much as possible! https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/appengine/ApplicationUrlDispatchRules.yaml is an example where we've done this previously.

File: mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml
Comment (rileykarson) [inline mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml:90]: nit: We may want to note that leaving the field unset is equivalent to sending an explicit `OFF`

File: mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml
Comment (rileykarson) [inline mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml:19]: Could you add a handwritten update test? One that transitions from `ENFORCED` -> `OFF `-> `UNENFORCED` (and then implicitly `OFF` again during delete) would cover things pretty exhaustively.

File: mmv1/templates/terraform/custom_delete/firebase_app_check_service_config.go
Comment (rileykarson) [inline mmv1/templates/terraform/custom_delete/firebase_app_check_service_config.go:1]: We should note in the resource docs that deleting the resource disables the enforcement mode!

File: mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml
Comment (weixifan) [inline mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml:21]: We don't support deletion at all for this resource, in case that matters.

File: mmv1/templates/terraform/custom_delete/firebase_app_check_service_config.go
Comment (weixifan) [inline mmv1/templates/terraform/custom_delete/firebase_app_check_service_config.go:32]: I don't know how to read this. What is the request exactly?

File: mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml
Comment (weixifan) [reply mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml:90]: In particular, we should note the existence of the `OFF` enum value being the default zero value.

File: mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml
Comment (rainshen49) [reply mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml:21]: Here, deletion will turn it OFF. I think this is the right behavior because OFF is the state before "creation". Another option is to treat deletion as a no-op, leaving the enforcement state as-is, and allowing an explicit OFF mode. WDYT? @weixifan

File: mmv1/templates/terraform/custom_delete/firebase_app_check_service_config.go
Comment (rainshen49) [reply mmv1/templates/terraform/custom_delete/firebase_app_check_service_config.go:32]: Yeah this is quite a mouthful. It roughly translates to 

`PATCH firebaseappcheck.googleapis.com/v1/projects/{{project}}/services/{{service_id}}?updateMask=enforcementMode
{} (empty body)

1. Method is PATCH
2. The `url` includes `updateMask` from line 16
3. The body `obj` is empty because it's never touched since declaration.

File: mmv1/templates/terraform/custom_create/firebase_app_check_service_config.go
Comment (rainshen49) [reply mmv1/templates/terraform/custom_create/firebase_app_check_service_config.go]: I forgot this old trick. Yes, this saves lots of code.

File: mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml
Comment (rainshen49) [reply mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml:20]: Added.

File: mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml
Comment (weixifan) [reply mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml:23]: Yes, that would work. Out of curiosity, which file contains the logic that "deletion will turn it OFF"?

Comment (rainshen49) [reply mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml:23]: With those two lines:
```yaml
delete_url: projects/{{project}}/services/{{service_id}}?updateMask=enforcementMode
delete_verb: :PATCH
```
and an absence of `custom_delete` in the `custom_code` section, the deletion request will be

```
PATCH firebaseappcheck.googleapis.com/{{project}}/services/{{service_id}}?updateMask=enforcementMode
{} (empty body)
```

Which is essentially turning it `OFF`.

**Review by rileykarson — APPROVED** [review]

File: mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml
Comment (rileykarson) [reply mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml]: This discussion thread looks resolved to me, but I'll wait for an ack from one of you before I click merge.

Comment (rainshen49) [reply mmv1/products/firebaseappcheck/firebase_app_check_service_config.yaml]: Yes this is good to merge. Leaving this block open so @weixifan can find it.

**Pull request merged**
